logstash-core/src/main/java/org/logstash/execution/Filter.java

```diff
@@ -0,0 +1,69 @@
+package org.logstash.execution;
+
+import org.logstash.Event;
+
+/**
+ * A Filter is simply a mapping of {@link QueueReader} to a new {@link QueueReader}.
+ */
+public interface Filter extends AutoCloseable {
+
+    QueueReader filter(QueueReader reader);
```

Comment:
The `QueueReader` isn't always a queue, obviously it could just be a previous filter. What about `PipelineReader` as a name?

Reply:
@andrewvc I have no opinion here tbh. We can go with `PipelineReader` :) Or maybe just `EventIterator`?

Reply:
I like the latter, `EventIterator` sounds good.